pkg/repository/manager.go

```diff
@@ -177,7 +180,55 @@ func (m *manager) PruneRepo(repo *velerov1api.BackupRepository) error {
 		return errors.WithStack(err)
 	}
 
-	return prd.PruneRepo(context.Background(), param)
+	log := m.log.WithFields(logrus.Fields{
+		"BSL name":  param.BackupLocation.Name,
+		"repo type": param.BackupRepo.Spec.RepositoryType,
+		"repo name": param.BackupRepo.Name,
+		"repo UID":  param.BackupRepo.UID,
+	})
+
+	log.Info("Start to maintence repo")
+
+	maintenanceJob, err := buildMaintenanceJob(m.maintenanceCfg, param, m.client, m.namespace)
+	if err != nil {
+		return errors.Wrap(err, "error to build maintenance job")
+	}
+
+	log = log.WithField("job", fmt.Sprintf("%s/%s", maintenanceJob.Namespace, maintenanceJob.Name))
+
+	if err := m.client.Create(context.TODO(), maintenanceJob); err != nil {
+		return errors.Wrap(err, "error to create maintenance job")
+	}
+	log.Debug("Creating maintenance job")
+
+	defer func() {
+		if err := deleteOldMaintenanceJobs(m.client, param.BackupRepo.Name,
+			m.maintenanceCfg.KeepLatestMaitenanceJobs); err != nil {
+			log.WithError(err).Error("Failed to delete maintenance job")
+		}
+	}()
+
+	var jobErr error
+	if err := waitForJobComplete(context.TODO(), m.client, maintenanceJob); err != nil {
+		log.WithError(err).Error("Error to wait for maintenance job complete")
+		jobErr = err // we won't return here for job may failed by maintenance failure, we want return the actual error
+	}
+
+	result, err := getMaintenanceResultFromJob(m.client, maintenanceJob)
+	if err != nil {
+		return errors.Wrap(err, "error to get maintenance job result")
+	}
+
+	if result != "" {
```

Comment:
Do we store the result in the repository CR's status?

Reply:
yes, we will store it in the message field of the status.